javalib/src/main/scala/java/lang/Integer.scala

```diff
@@ -50,127 +54,472 @@ object Integer {
+    if (negative) {
+      if (length == 1) throw new NumberFormatException(nm)
+      i += 1
+      first = nm.charAt(i)
+    }
 
-  @noinline def parseUnsignedInt(s: String, radix: scala.Int): scala.Int =
-    parseIntImpl(s, radix, signed = false)
+    var base = 10
+    if (first == '0') {
+      i += 1
+      if (i == length) return valueOf(0)
+      first = nm.charAt(i)
+      if (first == 'x' || first == 'X') {
+        i += 1
+        if (i == length) throw new NumberFormatException(nm)
+        base = 16
+      } else {
+        base = 8
+      }
+    } else if (first == '#') {
+      i += 1
+      if (i == length) throw new NumberFormatException(nm)
+      base = 16
+    }
 
-  @inline
-  private def parseIntImpl(s: String,
-                           radix: scala.Int,
-                           signed: scala.Boolean): scala.Int = ???
+    valueOf(parse(nm, i, base, negative))
+  }
 
-  @inline def toString(i: scala.Int): String =
-    ???
+  @inline
+  def divideUnsigned(dividend: scala.Int, divisor: scala.Int): scala.Int =
+    divUInt(dividend, divisor)
 
-  @inline def toUnsignedString(i: Int, radix: Int): String =
-    toStringBase(i, radix)
+  @inline def getInteger(nm: String): Integer =
+    getInteger(nm, null)
 
-  @inline def compare(x: scala.Int, y: scala.Int): scala.Int =
-    if (x == y) 0 else if (x < y) -1 else 1
-
-  @inline def compareUnsigned(x: scala.Int, y: scala.Int): scala.Int = ???
-
-  @inline def toUnsignedLong(x: Int): scala.Long =
-    x.toLong & 0xffffffffL
-
-  def bitCount(i: scala.Int): scala.Int = {
-    /* See http://graphics.stanford.edu/~seander/bithacks.html#CountBitsSetParallel
-     *
-     * The original algorithm uses *logical* shift rights. Here we use
-     * *arithmetic* shift rights instead. >> is shorter than >>>, especially
-     * since the latter needs (a >>> b) | 0 in JS. It might also be the case
-     * that >>> is a bit slower for that reason on some VMs.
-     *
-     * Using >> is valid because:
-     * * For the 2 first >>, the possible sign bit extension is &'ed away
-     * * For (t2 >> 4), t2 cannot be negative because it is at most the result
-     *   of 2 * 0x33333333, which does not overflow and is positive.
-     * * For the last >> 24, the left operand cannot be negative either.
-     *   Assume it was, that means the result of a >>> would be >= 128, but
-     *   the correct result must be <= 32. So by contradiction, it is positive.
-     */
-    val t1 = i - ((i >> 1) & 0x55555555)
-    val t2 = (t1 & 0x33333333) + ((t1 >> 2) & 0x33333333)
-    (((t2 + (t2 >> 4)) & 0xF0F0F0F) * 0x1010101) >> 24
+  @inline def getInteger(nm: String, v: scala.Int): Integer = {
+    val result = getInteger(nm, null)
+    if (result == null) new Integer(v)
+    else result
   }
 
-  @inline def divideUnsigned(dividend: Int, divisor: Int): Int =
-    divUInt(dividend, divisor)
+  def getInteger(nm: String, v: Integer): Integer =
+    if (nm == null || nm.length() == 0) {
+      valueOf(v)
```

Comment:
Fixed in https://github.com/scala-native/scala-native/pull/238/commits/f0eab14d727ad8978b63a07733f4f4a89798198f